### PR TITLE
Add HDM to oss compose

### PR DESCRIPTION
**Pull request by tuxmea**
issue: we need to have at least one agent connected to puppetserver with entries in puppetdb.

going to copy the parts from ha

### CODE REVIEW

File: puppet/oss/.env [hunk 1]
```diff
@@ -1,4 +1,5 @@
 REGISTRY_PATH = "ghcr.io/voxpupuli"
+HDM_REGISTRY_PATH = "ghcr.io/betadots"
```
Comment (rwaffen) [inline puppet/oss/.env:2]: please don't use this pattern, i want to remove it. when doing this, renovate cannot scan for new images

File: puppet/oss/compose.yaml [hunk 2]
```diff
@@ -39,6 +40,7 @@ services:
     restart: always
     ports:
       - 8081:8081
+      - 8080:8080
```
Comment (rwaffen) [inline puppet/oss/compose.yaml:43]: you don't need to expose the port, if you don't want to use it from the outside of this stack. the container to each other can see the open ports without an explicit expose.

File: puppet/oss/compose.yaml [hunk 3]
```diff
@@ -84,6 +86,34 @@ services:
     depends_on:
       - puppetdb
 
+  hdm:
+    image: ${HDM_REGISTRY_PATH}/hdm:latest
```
Comment (rwaffen) [inline puppet/oss/compose.yaml:90]: please do full path hier, like mentioned above.

File: puppet/oss/compose.yaml [hunk 4]
```diff
@@ -84,6 +86,34 @@ services:
...
+      - hdm
+
+  testing:
+    image: ghcr.io/betadots/pdc:latest
+    container_name: crafty-testing
+    hostname: crafty-testing
+    platform: linux/x86_64
+    volumes:
+      - ./puppet.conf:/etc/puppetlabs/puppet/puppet.conf:ro
+      - ./agent-ssl:/etc/puppetlabs/puppet/ssl
```
Comment (rwaffen) [inline puppet/oss/compose.yaml:112]: please add the agent-ssl to gitignore, like in the ha sample

File: puppet/oss/compose.yaml [hunk 5]
```diff
@@ -84,6 +86,34 @@ services:
...
+    profiles:
+      - hdm
+
+  testing:
+    image: ghcr.io/betadots/pdc:latest
+    container_name: crafty-testing
+    hostname: crafty-testing
+    platform: linux/x86_64
+    volumes:
+      - ./puppet.conf:/etc/puppetlabs/puppet/puppet.conf:ro
```
Comment (rwaffen) [inline puppet/oss/compose.yaml:111]: where is the puppet.conf coming from? it is not in this pr?

**Review by rwaffen — CHANGES_REQUESTED** [review]
please also add to the readme how to use this. maybe we should add a dedicated readme like in the ha example 🤔

**Review by rwaffen — APPROVED** [review]

**Comment by tuxmea** [comment]
Looks good to me. As I am the author of this PR, I am not able to approve.

**Pull request merged**